roofit/roofitcore/inc/RooDataHist.h

```diff
@@ -106,6 +106,7 @@ class RooDataHist : public RooAbsData, public RooDirItem {
                const std::map<const RooAbsArg*, std::pair<double, double> >& ranges,
                std::function<double(int)> getBinScale = [](int){ return 1.0; } );
 
+  void weights(double* output, size_t nEvents, RooAbsArg* pdfObsList, RooFit::Detail::DataMap const& dataMap, bool correctForBinSize);
```

Comment:
```suggestion
  void weights(double* output, RooSpan<double const> xVals, bool correctForBinSize);
```
I would suggest another interface here. Instead of passing the `RooAbsArg` and the dataMap, just pass the element of the dataMap which is of type `RooSpan<double const>` (you can pass a copy because it's a very small object). You also don't need to pass the number of events, because it's encoded in the size of the span.

By the way, a RooSpan is almost exactly like a `std::span`:
https://en.cppreference.com/w/cpp/container/span

It is like a pointer to the beginning of an array together with size information.

Reply:
This is a nice way to do it, combined with ```_lvbins```! Thanks!